api/observability/v1/filter_types.go

```diff
@@ -41,10 +41,15 @@ var (
 )
 
 // FilterSpec defines a filter for log messages.
+// +kubebuilder:validation:XValidation:rule="self.type != 'kubeAPIAudit' || has(self.kubeAPIAudit)", message="Additional type specific spec is required for the filter type"
```

Comment:
nit: Maybe make sense to add filter type name to the validation message to be more clear